### PR TITLE
[JENKINS-71200] Fix elements property overload in ListView

### PR DESCRIPTION
**Pull request by Dohbedoh**
See [JENKINS-71200](https://issues.jenkins.io/browse/JENKINS-71200).

`ListView` produces `<input ..>` elements for each items within the ItemGroup that are named directly after the item name itself. If an item name is `elements`, this kind of overloads the `form.elements` attribute which causes a bad form submission.

The propose fix is to prefix the input names with `item_` to avoid the overload. 

### Testing done

* Started Jenkins
* create an freestyle job named `elements`
* create a view and Save

### Proposed changelog entries

- Prefix the name of input elements of ListView to prevent form submission issues when an Item (job) is named `elements`.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7942"></a>

### CODE REVIEW

**Comment by Dohbedoh** [comment]
First commit is a test that reproduces the problem and show the failure.
Second commit is the fix.

**Comment by timja** [comment]
@NotMyFault / @Dohbedoh was this introduced in 2.403?

**Comment by Dohbedoh** [comment]
@timja I don't think so. I find this out on an instance running 2.361.3. And I could reproduce this although in the latest weekly 2.403. This has probably been around for a while.

**Comment by NotMyFault** [comment]
> @NotMyFault / Dohbedoh was this introduced in 2.403?

Unlikely, I just labeled it as regression because I recall it working at some point.

**Comment by timja** [comment]
I doubt this ever worked

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by daniel-beck** [comment]
> I doubt this ever worked

FTR this hasn't worked in many years, if ever (at least for some attribute names), see #6959 / JENKINS-10693.

File: core/src/main/java/hudson/model/ListView.java
Comment (basil) [inline core/src/main/java/hudson/model/ListView.java:455]: https://github.com/jenkinsci/categorized-view-plugin/pull/38

**Comment by basil** [comment]
Caused [JENKINS-71403](https://issues.jenkins.io/browse/JENKINS-71403).

**Comment by basil** [comment]
https://github.com/jenkinsci/calendar-view-plugin/blob/7cdb173e5890913c9297d6f9e0bc59cc7ec817bf/src/main/resources/io/jenkins/plugins/view/calendar/CalendarView/configure-entries.jelly#L292 needs to be adapted as well.

**Pull request merged**
